server/game/cards/02_SHD/events/CalculatedLethality.ts

```diff
@@ -24,14 +24,13 @@ export default class CalculatedLethality extends EventCard {
                     AbilityHelper.immediateEffects.defeat()
                 ])
             },
-            ifYouDo: (ifYouDoContext) => ({
+            then: (thenContext) => ({
```

Comment:
This currently might fail if no target is available for defeat, since `thenContext.targets.upgradeAmount` will not be set. I think we need to add a `thenCondition` checking if no target was selected at all, and we should have a test confirming it works.